webapp/src/shared/components/Button/Button.tsx

```diff
@@ -0,0 +1,43 @@
+import { ReactNode } from 'react';
+import './Button.css';
+
+export enum ButtonTypes {
+  SUBMIT = 'submit',
+  WARNING = 'warning',
+}
+
+type Buttonprops = {
+  buttonType: ButtonTypes;
+  buttonIcon?: JSX.Element;
+  action?: () => void;
+  disabled?: boolean;
+  children: ReactNode;
+};
+
+export default function Button({
+  buttonType,
+  buttonIcon,
+  action,
+  disabled = false,
+  children,
+}: Buttonprops) {
+  let buttonClassName = 'button text-style-3-bold';
+  if (buttonType === ButtonTypes.SUBMIT) {
+    buttonClassName += ' button-submit';
+  } else if (buttonType === ButtonTypes.WARNING) {
+    buttonClassName += ' button-warning';
+  }
+  if (buttonIcon) {
+    buttonClassName += ' button-icon';
+  }
+  if (disabled) {
+    buttonClassName += ' button-disabled';
+  }
+
+  return (
+    <button className={buttonClassName} disabled={disabled} onClick={action}>
+      {buttonIcon}
+      {children}
+    </button>
```

Comment:
I'll just do the following:

```tsx
export default function Button({
  buttonType,
  buttonIcon,
  action,
  disabled = false,
  children,
}: Buttonprops) {
  return (
    <button
      className={`text-style-3-bold button button-${buttonType}`}
      disabled={disabled}
      onClick={action}
    >
      {buttonIcon && <div className="button-icon">{buttonIcon}</div>}
      {children}
    </button>
  );
}
```

and css:

```css
button:disabled {
  opacity: 0.5;
  cursor: default;
}

.button {
  width: 312px;
  height: 58px;
  border-radius: 8px;
  color: var(--color-light);
  border: 0;
  position: relative;
  align-items: center;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.button-icon {
    display: flex;
    position: absolute;
    left: 20px;
}

/* rest of classes, variants etc ....*/
```

This way:

- Get rid of if-else class selection by:
  1. Using a selector for `disable` state.
  2. Inline template literal to select button class variant
  3. Inline validation for iconComponent (so, if it does not exist, we don't draw it)
- Use flex for the button to center the text and fix the icon to the left of the button. If you prefer to use grid, I'd create a grid in the main `button` class so text and icon have a defined position.

Reply:
Nice! I like it! And regarding grid or <div>, I don't care using one or another, so I leave it as you proposed.